examples/fabric-admin/scripts/fabric-sync-app.py

```diff
@@ -0,0 +1,251 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2024 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import contextlib
+import os
+import signal
+import sys
+from argparse import ArgumentParser
+from tempfile import TemporaryDirectory
+
+BRIDGE_COMMISSIONED = asyncio.Event()
+# Log message which should appear in the fabric-admin output if
+# the bridge is already commissioned.
+BRIDGE_COMMISSIONED_MSG = b"Reading attribute: Cluster=0x0000_001D Endpoint=0x1 AttributeId=0x0000_0000"
+
+
+async def forward_f(f_in, f_out, prefix: str):
+    """Forward f_in to f_out with a prefix attached."""
+    global BRIDGE_COMMISSIONED
+    while True:
+        line = await f_in.readline()
+        if not line:
+            break
+        if not BRIDGE_COMMISSIONED.is_set() and BRIDGE_COMMISSIONED_MSG in line:
+            BRIDGE_COMMISSIONED.set()
+        f_out.buffer.write(prefix.encode())
```

Comment:
nit: why doesn't the caller provide the encoded string in? Or could we just do the encode once before the while loop?

Could we add types for f_in/f_out? how do we know they have a `buffer` member and a flush one?

Reply:
We seem to want `https://docs.python.org/3/library/typing.html#typing.BinaryIO` since we encode bytes.